### DISCUSSION TITLE
Does PyPSA support using RL for optimizing battery scheduling and economic dispatch?

### DISCUSSION THREAD

**Discussion by karima9119** [PyPSA/PyPSA#782] [Q&A]
Hello everyone,

I'm currently exploring the optimization of battery scheduling in an islanded microgrid using reinforcement learning (RL). My project involves the management of an islanded microgrid that includes two generators, necessitating stability, and commitment considerations.

I'm particularly interested in utilizing PyPSA for this purpose. Could someone please advise me on whether PyPSA is capable of supporting such a scenario? Essentially, I aim to use PyPSA to model the microgrid as a plant for simulation, enabling me to inject inputs for the governors and read relevant variables.

**Comment by fneum** [comment]
@karima9119, you can generally use PyPSA and its data structure for this application. However, PyPSA does not use any machine learning techniques for optimisation but solves LP/MILP problems by interfacing to solvers.

**Reply by karima9119** [reply]
All I need is to interface my own _solver_ (based on RL) with a system as described, in islanded mode.
